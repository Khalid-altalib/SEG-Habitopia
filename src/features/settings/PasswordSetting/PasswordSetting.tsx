import { Button, Input, Text, View } from "native-base";
import React, { useState } from "react";
import { Controller, useForm } from "react-hook-form";
import DoneButton from "../DoneButton.tsx/DoneButton";

type Props = {};

const PasswordSetting = (props: Props) => {
  const { control, getValues, watch } = useForm({
    defaultValues: {
      formValue: "",
      confirmPassword: "",
    },
  });

  watch();

  const arePasswordsNotEqual = () =>
    getValues("formValue") !== getValues("confirmPassword");

  return (
    <View>
      <View marginBottom={5}>
        <Text mb={5}>New Password</Text>
        <Controller
          control={control}
          render={({ field: { onChange, value } }) => (
            <Input onChangeText={onChange} value={value} size="xl" />
          )}
          name="formValue"
        />
      </View>
      <View>
        <Text mb={5}>Confirm Password</Text>
        <Controller
          control={control}
          render={({ field: { onChange, value } }) => (
            <Input onChangeText={onChange} value={value} size="xl" />
          )}
          name="confirmPassword"
        />
      </View>
      <DoneButton
        getValues={getValues}
        valueName="password"
        disabled={arePasswordsNotEqual()}
      />
    </View>
  );
};

export default PasswordSetting;
