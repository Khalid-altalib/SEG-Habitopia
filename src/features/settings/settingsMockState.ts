import { SettingsState } from "./settingsSlice";

/**
 * The mock state for the settings redux state
 */

const settingsMockState: SettingsState = {
  settings: {
    email: "test@test.com",
    password: "password",
    notifications: true,
    name: "Test User",
    biography: "Hello, I'm Test User.",
  },
  fetchSettings: {
    loading: false,
    error: "",
  },
  setSettings: {
    loading: false,
    error: "",
  },
};

export default settingsMockState;
