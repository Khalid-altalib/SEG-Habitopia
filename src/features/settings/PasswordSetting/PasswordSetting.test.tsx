import { render } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import PasswordSetting from "./PasswordSetting";
import settingsMockState from "../settingsMockState";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";

describe("PasswordSetting", () => {
  const mockState = {
    settings: settingsMockState,
  };
  const mockStore = configureStore([thunk])(mockState);

  it("renders correctly", () => {
    const tree = render(
      <TestingWrapperNavigation store={mockStore}>
        <PasswordSetting />
      </TestingWrapperNavigation>
    );

    expect(tree).toMatchSnapshot();
  });
});
