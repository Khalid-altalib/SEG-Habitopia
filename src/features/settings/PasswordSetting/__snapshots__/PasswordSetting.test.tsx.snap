// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordSetting renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <View
                dataSet={{}}
                style={
                  {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "marginBottom": 20,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Old Password
                </Text>
                <View
                  dataSet={{}}
                  isFocused={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderColor": "#d4d4d4",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "overflow": "hidden",
                    }
                  }
                >
                  <TextInput
                    accessible={true}
                    dataSet={{}}
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholderTextColor="#a3a3a3"
                    secureTextEntry={false}
                    style={
                      {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 18,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      }
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "marginBottom": 20,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  New Password
                </Text>
                <View
                  dataSet={{}}
                  isFocused={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderColor": "#d4d4d4",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "overflow": "hidden",
                    }
                  }
                >
                  <TextInput
                    accessible={true}
                    dataSet={{}}
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholderTextColor="#a3a3a3"
                    secureTextEntry={false}
                    style={
                      {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 18,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      }
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                dataSet={{}}
                style={{}}
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "marginBottom": 20,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Confirm Password
                </Text>
                <View
                  dataSet={{}}
                  isFocused={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderColor": "#d4d4d4",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "overflow": "hidden",
                    }
                  }
                >
                  <TextInput
                    accessible={true}
                    dataSet={{}}
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholderTextColor="#a3a3a3"
                    secureTextEntry={false}
                    style={
                      {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 18,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      }
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                isFullWidth={true}
                isLoading={false}
                onPress={[Function]}
                style={
                  {
                    "marginTop": 25,
                  }
                }
                testID="button"
                type={0}
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "marginTop": 25,
                      "opacity": 1,
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    proxiedProperties={
                      {
                        "colors": [
                          4278220774,
                          4286456526,
                        ],
                        "endPoint": [
                          1,
                          1,
                        ],
                        "locations": undefined,
                        "startPoint": [
                          0,
                          0,
                        ],
                      }
                    }
                    style={
                      [
                        {
                          "alignItems": "center",
                          "borderRadius": 10,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "paddingHorizontal": 40,
                          "paddingVertical": 15,
                        },
                        {
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "color": "#e4e4e7",
                            "fontFamily": undefined,
                            "fontSize": 19,
                          },
                          [
                            null,
                            {
                              "alignSelf": "center",
                            },
                          ],
                        ]
                      }
                    >
                      Done
                    </Text>
                    <View
                      style={
                        {
                          "marginLeft": 5,
                        }
                      }
                    />
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
